src/Credentials/AssumeRoleWithWebIdentityCredentialProvider.php

```diff
@@ -94,12 +94,13 @@ public function __invoke()
             $result = null;
             while ($result == null) {
                 try {
-                    $token = file_get_contents($this->tokenFile);
-                    if (false === $token) {
+                    if (is_readable($this->tokenFile)) {
+                        $token = file_get_contents($this->tokenFile);
+                    } else {
                         clearstatcache(true, dirname($this->tokenFile) . "/" . readlink($this->tokenFile));
                         clearstatcache(true, dirname($this->tokenFile) . "/" . dirname(readlink($this->tokenFile)));
                         clearstatcache(true, $this->tokenFile);
-                        $token = file_get_contents($this->tokenFile);   
+                        $token = file_get_contents($this->tokenFile);
                     }
                 } catch (\Exception $exception) {
```

Comment:
Can you add a test case for the logic?

Reply:
The logic is a bit different now from when you wrote this; a test case was still added to ensure that the error was not thrown